src/coreclr/jit/fgopt.cpp

```diff
@@ -788,6 +788,16 @@ unsigned Compiler::fgDfsReversePostorder()
     //
     fgDfsReversePostorderHelper(fgFirstBB, visited, preorderIndex, postorderIndex);
 
+    // For OSR, walk from the original method entry too.
+    //
+    if (opts.IsOSR() && (fgEntryBB != nullptr))
+    {
+        if (!BlockSetOps::IsMember(this, visited, fgEntryBB->bbNum))
+        {
+            fgDfsReversePostorderHelper(fgEntryBB, visited, preorderIndex, postorderIndex);
```

Comment:
These aren't really reachable, right? (Except for maybe tailcall-to-loop opt). But we want them in the order to be able to propagate facts from them.

I wonder if SSA/VN could benefit similarly to seeing these.

Reply:
It seems like a sign that CQ for OSR methods would benefit from inserting the jump to the start block very late, e.g. right before lowering (probably with some additional DCE for the unreachable blocks). Of course at some cost of TP, so maybe it would need to be restricted to loop-based patchpoints while the current behavior is ideal for partial compilation.

Reply:
Added a note to#33658 to try this someday. One consideration for deferral is that we might need to alter or block some optimizations that might remove computations from the OSR side (hoisting, CSE).

Reply:
Also note currently `fgEntryBB` gets nulled out after morph.